### PR TITLE
feat(DatePicker): make onChange and value required

### LINKED ISSUE #430: [DatePicker] Make value and onChange props required
The uncontrolled version of our DatePicker is very buggy and not very useful. I think those two props should be required so the component is always controlled.

### PR DESCRIPTION
**Pull request by tiagoalvesdulce**
Closes #430

### CODE REVIEW

File: src/components/Datepicker/Datepicker.jsx [hunk 1]
```diff
@@ -385,10 +385,22 @@ DatePicker.propTypes = {
...
   isRange: PropTypes.bool,
   lang: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
-  onChange: PropTypes.func,
+  onChange: function (props, propName) {
+    var fn = props[propName];
+    if (
+      !fn.prototype ||
+      typeof fn.prototype.constructor !== "function" ||
+      (fn.prototype.constructor.length < 2 &&
+        fn.prototype.constructor.length > 3)
```
Comment (victorgcramos) [inline src/components/Datepicker/Datepicker.jsx:397]: I think the operator here is `||` 😁

File: src/components/Datepicker/Datepicker.jsx [hunk 2]
```diff
@@ -385,10 +385,22 @@ DatePicker.propTypes = {
...
   ]),
-  value: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
+  value: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
   isRange: PropTypes.bool,
   lang: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
-  onChange: PropTypes.func,
+  onChange: function (props, propName) {
+    var fn = props[propName];
+    if (
+      !fn.prototype ||
```
Comment (victorgcramos) [inline src/components/Datepicker/Datepicker.jsx:394]: when `onChange` is `null` or `undefined`, the error message is:
```error
Warning: Failed prop type: Cannot read properties of undefined (reading 'prototype')
```

I think we should also display the `... must be a function with 2 (if isMonthsMode is on) or 3 args. Args are: year, month and day.` error.

**Review by victorgcramos — CHANGES_REQUESTED** [review]
Left some inline comments! Thanks!

File: src/components/Datepicker/Datepicker.jsx (hunk 1 above)
Comment (tiagoalvesdulce) [reply src/components/Datepicker/Datepicker.jsx:397]: lol. right!

File: src/components/Datepicker/Datepicker.jsx (hunk 2 above)
Comment (tiagoalvesdulce) [reply src/components/Datepicker/Datepicker.jsx:394]: Yeah, will add that. thanks!

**Review by victorgcramos — APPROVED** [review]
LGTM!

**Pull request merged**